ExcelPlotter.psm1

```diff
@@ -28,68 +28,75 @@ function Create-ExcelFile {
-                                -row $rowOffset -iter $iter -numIters $numIters
-        $null = Fill-RowLabels -Worksheet $worksheetObject -rows $table.rows -startRow ($table.meta.colLabelDepth + $rowOffset) `
-                                -col 1 -iter $iter -numIters $numIters
-        $null = Fill-Data -Worksheet $worksheetObject -Data $table.data -Cols $table.cols -Rows $table.rows `
-                                -StartCol ($table.meta.rowLabelDepth + 1) -StartRow ($table.meta.colLabelDepth + $rowOffset) `
-                                -iter $iter -numIters $numIters
-        
-        if ($table.chartSettings) { 
-            $null = Create-Chart -Worksheet $worksheetObject -Table $table -StartCol 1 -StartRow $rowOffset -chartNum $chartNum
-            $chartNum += 1
-        } 
-        Format-ExcelSheet -Worksheet $worksheetObject -Table $table -RowOffset $rowOffset
-        $rowOffset += $table.meta.colLabelDepth + $table.meta.dataHeight + 1
-    }
+            $null = Fill-ColLabels -Worksheet $worksheetObject -cols $table.cols -startCol ($table.meta.rowLabelDepth + 1) `
+                                    -row $rowOffset -iter $iter -numIters $numIters
+            $null = Fill-RowLabels -Worksheet $worksheetObject -rows $table.rows -startRow ($table.meta.colLabelDepth + $rowOffset) `
+                                    -col 1 -iter $iter -numIters $numIters
+            $null = Fill-Data -Worksheet $worksheetObject -Data $table.data -Cols $table.cols -Rows $table.rows `
+                                    -StartCol ($table.meta.rowLabelDepth + 1) -StartRow ($table.meta.colLabelDepth + $rowOffset) `
+                                    -iter $iter -numIters $numIters
+            
+            if ($table.chartSettings) { 
+                $null = Create-Chart -Worksheet $worksheetObject -Table $table -StartCol 1 -StartRow $rowOffset -chartNum $chartNum
+                $chartNum += 1
+            } 
+            Format-ExcelSheet -Worksheet $worksheetObject -Table $table -RowOffset $rowOffset
+            $rowOffset += $table.meta.colLabelDepth + $table.meta.dataHeight + 1
+        }
 
-    Fit-Cells -Worksheet $worksheetObject
-    Write-Progress -Activity "Creating Excel File" -Status "Done" -Id 4 -PercentComplete 100
-    $null = $workbookObject.SaveAs($SavePath, [Excel.XlFileFormat]::xlOpenXMLWorkbook)
-    $workbookObject.Saved = $true
-    $null = $workbookObject.Close()
-    $null = [System.Runtime.Interopservices.Marshal]::ReleaseComObject($workbookObject)
-
-    $null = $excelObject.Quit()
-    $null = [System.Runtime.Interopservices.Marshal]::ReleaseComObject($excelObject)
-    $null = [System.GC]::Collect()
-    $null = [System.GC]::WaitForPendingFinalizers()
+        Fit-Cells -Worksheet $worksheetObject
+        Write-Progress -Activity "Creating Excel File" -Status "Done" -Id 4 -PercentComplete 100
+    } catch {
+        Write-Warning "Exception $($_.Exception.Message) in $($MyInvocation.MyCommand.Name)"
```

Comment:
Replace with `throw $_`, no need for anything else in the catch block. This will cause the original error to be rethrown after the finally block, preserving the original error and context.